internal/app/handlers/handlers.go

```diff
@@ -0,0 +1,65 @@
+package handlers
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"shorty/internal/app/config"
+	"shorty/internal/app/hash"
+	"shorty/internal/app/storage"
+)
+
+func Shortify(writer http.ResponseWriter, request *http.Request, cfg config.Config, str *storage.Storage) {
+	if request.Method != http.MethodPost {
+		http.Error(writer, "Only POST requests are allowed", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(request.Body)
+	if err != nil {
+		http.Error(writer, "Internal server error", http.StatusInternalServerError)
+		log.Println("failed to parse request body")
```

Comment:
В Go принято логировать и саму ошибку, чтобы информация о причине не терялась:

```suggestion
		log.Printf("failed to parse request body: %v", err)
```